code/game/objects/structures/bounty_post.dm

```diff
@@ -0,0 +1,268 @@
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE
+	///maximum amount of bounties we can hold
+	var/bounty_amount = 8
+
+/obj/structure/bounty_post/Initialize(mapload)
+	. = ..()
+	setup_bounties()
+
+/obj/structure/bounty_post/proc/setup_bounties()
+	var/bounties_to_set = bounty_amount - length(GLOB.holy_contracts)
+	for(var/count in 1 to bounties_to_set)
+		new /datum/holy_bounty
+	addtimer(CALLBACK(src, PROC_REF(setup_bounties)), 10 MINUTES) //every 10 minutes, regenerate bounties
+
+/obj/structure/bounty_post/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "bountypost")
+		ui.open()
+
+/obj/structure/bounty_post/ui_data(mob/user)
+	var/list/data = list()
+	data["bounty"] = list()
+	for(var/datum/holy_bounty/our_bounty as anything in GLOB.holy_contracts)
+		if(!isnull(our_bounty.assignee?.resolve()))
+			continue //dont display bounties that are already claimed
+		data["bounty"] += list(list(
+			"bounty_id" = REF(our_bounty),
+			"bounty_name" = our_bounty.bounty_name,
+			"bounty_icon" = icon2base64(our_bounty.bounty_icon),
+			"bounty_reward" = our_bounty.reward_points,
+		))
+	data["user_points"] = GLOB.bounty_points_tracker[REF(user)] || 0
+	return data
+
+/obj/structure/bounty_post/ui_static_data(mob/user)
+	var/list/data = list()
+	data["shop_item"] = list()
+	for(var/datum/bounty_store_item/shop_item as anything in subtypesof(/datum/bounty_store_item))
+		var/obj/item/item_path = shop_item::item_path
+		data["shop_item"] += list(list(
+			"item_ref" = shop_item::type,
+			"item_name" = item_path::name,
+			"item_price" = shop_item::item_price,
+			"item_icon" = icon2base64(getFlatIcon(image(icon = item_path::icon, icon_state = item_path::icon_state), no_anim=TRUE)),
+			"item_description" = item_path::desc,
+		))
+	data["coins_icon"] = icon2base64(icon('icons/obj/mining.dmi', "shop_coins"))
+	return data
+
+/obj/structure/bounty_post/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(. || !isliving(ui.user))
+		return TRUE
+	var/mob/living/user = ui.user
+	switch(action)
+		if("claim")
+			var/datum/holy_bounty/selected_bounty
+			if(isnull(params["reference"]))
+				return TRUE
+			for(var/datum/holy_bounty/bounty as anything in GLOB.holy_contracts)
+				if(params["reference"] != REF(bounty))
+					continue
+				selected_bounty = bounty
+				break
+			if(isnull(selected_bounty))
+				return TRUE
+			if(selected_bounty.assignee?.resolve())
+				to_chat(user, span_warning("This bounty has already been assigned to someone!"))
+				return TRUE
+			if(!selected_bounty.assign_bounty(user))
+				return TRUE
+			var/obj/item/bounty_contract/contract = new()
+			contract.set_bounty(selected_bounty)
+			user.put_in_hands(contract)
+			return TRUE
+		if("purchase")
+			var/datum/bounty_store_item/reward_path = text2path(params["reference"])
+			if(isnull(reward_path))
+				return TRUE
+			if(!(reward_path in subtypesof(/datum/bounty_store_item)))
```

Comment:
`ispath(reward_path, /datum/bounty_store_item)`

Reply:
my mind was fried by this point